breacharbiter.go

```diff
@@ -594,17 +596,287 @@ func (b *breachArbiter) createJusticeTx(r *retributionInfo) (*wire.MsgTx, error)
 	// witnesses for both commitment outputs, and all the pending HTLCs at
 	// this state in the channel's history.
 	// TODO(roasbeef): handle the 2-layer HTLCs
-	localWitness, err := r.selfOutput.witnessFunc(justiceTx, hashCache, 0)
+	localWitnessFunc := r.selfOutput.witnessType.GenWitnessFunc(
+		&b.wallet.Signer, r.selfOutput.signDescriptor)
+	localWitness, err := localWitnessFunc(justiceTx, hashCache, 0)
 	if err != nil {
 		return nil, err
 	}
 	justiceTx.TxIn[0].Witness = localWitness
 
-	remoteWitness, err := r.revokedOutput.witnessFunc(justiceTx, hashCache, 1)
+	remoteWitnessFunc := r.revokedOutput.witnessType.GenWitnessFunc(
+		&b.wallet.Signer, r.revokedOutput.signDescriptor)
+	remoteWitness, err := remoteWitnessFunc(justiceTx, hashCache, 1)
 	if err != nil {
 		return nil, err
 	}
 	justiceTx.TxIn[1].Witness = remoteWitness
 
 	return justiceTx, nil
 }
+
+// breachedOutput contains all the information needed to sweep a breached
+// output. A breached output is an output that we are now entitled to due to a
+// revoked commitment transaction being broadcast.
+type breachedOutput struct {
+	amt      btcutil.Amount
+	outpoint wire.OutPoint
+
+	signDescriptor *lnwallet.SignDescriptor
+	witnessType    lnwallet.WitnessType
+
+	twoStageClaim bool
+}
+
+// retribution encapsulates all the data needed to sweep all the contested
```

Comment:
Why the name change? Unnecessarily adds to the diff. If there isn't a particular reason, then the prior name should be reverted.